src/domain/usecases/metadata/GetSharingSummaryUseCase.ts

```diff
@@ -0,0 +1,99 @@
+import _ from "lodash";
+import { FutureData } from "../../entities/Future";
+import { MetadataRepository } from "../../repositories/MetadataRepository";
+import { MetadataPayload } from "../../entities/MetadataItem";
+import { SharingWarning, SharingSummary, SharingPayload } from "../../entities/SharingSummary";
+
+export class GetSharingSummaryUseCase {
```

Comment:
To get summary data you have to take into account also excluded data from step 2 and global excluded data, because these data it's not going to be changed

Reply:
In this use case you have to compare sharing settings of children vs sharing settings of parent to check if there is difference, but if the user is going to update that sharing settings then instead of compare the sharing settings that come from dhis2 (the current shering settting of metadata) you have to use the new sharing settings that are going to be applied.